pyjob/script.py

```diff
@@ -288,6 +288,17 @@ def write(self):
             f_out.write(str(self))
         os.chmod(fname, 0o777)
 
+    def cleanup(self):
```

Comment:
should this be called even if execution failed?

Reply:
The files won't get deleted if they don't exist, which means it should be fine if execution fails; unless the user wants to check the log for debug, but I guess then the user shouldn't ask to delete these files. At the moment I can't see any property to store if the execution of the script failed, I guess I would need add it and then only cleanup if it the exit code was ok. I'm not sure if this would be necessary considering it's an optional behaviour... The one thing I'm concerned about is that `cleanup()` gets called by `pyjob.ClusterTask.close()`, which is also called by the context manager at `__exit__`, like for example at the end of the python script / execution. What do you think about this?

Reply:
ok. let's leave for now. need to think about this one again tomorrow.